unixfs/io/pbdagreader.go

```diff
@@ -181,6 +181,11 @@ func (dr *PBDagReader) CtxReadFull(ctx context.Context, b []byte) (int, error) {
 			return total, nil
 		}
 
+		// We haven't hit the end yet.
+		if err != io.EOF {
+			continue
+		}
+
```

Comment:
@Stebalien what do you think about using `io.ReadFull` or `io.ReadAtLeast` here. They will return `io.ErrUnexpectedEOF` if EOF is hit before the buffer is full and thus telling us to load the next buffer.

This would decrease significantly complexity of this function.

For context: https://golang.org/src/io/io.go?s=10738:10804#L293

Reply:
I considered it but then I got lazy. However, you are right. It'll probably make this simpler.

Reply:
So, this function actually needs to read *multiple* files so using ReadFull ends up making it more complex.